docs/basics.md

```diff
@@ -0,0 +1,203 @@
+  * [Value Attributes](#value-attributes)
+  * [Functions Returning Values](#functions-returning-values)
+* [Setting Options](#setting-options)
+  * [The `getOption` Method](#the-getoption-method)
+  * [The `mergeOptions` Method](#the-mergeoptions-method)
+
+## Class-based Inheritance
+
+Backbone and Marionette utilize the `_.extend` function to simulate class-based
+inheritance. All built-in classes, such as `Marionette.View`,
+`Marionette.Object` and everything that extend these provide an `extend` method
+for just this purpose.
+
+In the below example, we create a new pseudo-class called `MyView`:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({});
+```
+
+You can now create instances of `MyView` with JavaScript's `new` keyword:
+
+```javascript
+var view = new MyView();
+```
+
+### Value Attributes
+
+When we extend classes, we can set a form of class attributes by setting new
+values to be a specific value by setting it when we extend our class:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({
+  className: 'bg-success',
+
+  template: '#template-identifier',
+
+  regions: {
+    myRegion: '.my-region'
+  },
+
+  modelEvents: {
+    change: 'removeBackground'
+  },
+
+  removeBackground: function() {
+    this.$el.removeClass('bg-success');
+  }
+});
+```
+
+When we instantiate `MyView`, each instance will be given a `.bg-success` class
+with a `myRegion` region created on the `.my-region` element.
+
+### Functions Returning Values
+
+In almost every instance where we can set a value, we can also assign a function
+to figure out the value at runtime. In this case, Marionette will run the
+function on instantiation and use the returned value:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({
+  className: function() {
+    return this.model.successful() ? 'bg-success' : 'bg-error';
+  },
+
+  template: '#template-identifier',
+
+  regions: function() {
+    return {
+      myRegion: '.my-region'
+    };
+  },
+
+  modelEvents: function() {
+    var wasSuccessful = this.model.successful();
+    return {
+      change: wasSuccessful ? 'removeBackground' : 'alert'
+    };
+  },
+
+  removeBackground: function() {
+    this.$el.removeClass('bg-success');
+  },
+
+  alert: function() {
+    window.alert('model changed');
+  }
+});
+```
+
+As we can see, almost all of the attributes here can be worked out dynamically.
+In most cases, Marionette will call the function once at instantiation, or first
+render, and preserve the value throughout the lifetime of the View. There are
+some exceptions to this rule - these will be referred to with their respective
+documentation.
+
+There are some exceptions to this general rule that will be documented in the
+relevant places.
+
+### Function Context
+
+Each function call will receive the class instance as its `this` context - this
```

Comment:
The formatting looks off.
> `this` context - this

or

> `this` context
- this gives you a handy way to ...

or

> `this` context - this gives you a handy way to ...

?